web/components/contract/quick-bet.tsx

```diff
@@ -0,0 +1,242 @@
+import clsx from 'clsx'
+import { getOutcomeProbability, getTopAnswer } from 'common/calculate'
+import { getExpectedValue } from 'common/calculate-dpm'
+import {
+  Contract,
+  FullContract,
+  CPMM,
+  DPM,
+  Binary,
+  NumericContract,
+  FreeResponse,
+} from 'common/contract'
+import { formatMoney } from 'common/util/format'
+import toast from 'react-hot-toast'
+import { useUser } from 'web/hooks/use-user'
+import { useUserContractBets } from 'web/hooks/use-user-bets'
+import { placeBet } from 'web/lib/firebase/api-call'
+import { getBinaryProb } from 'web/lib/firebase/contracts'
+import TriangleDownFillIcon from 'web/lib/icons/triangle-down-fill-icon'
+import TriangleFillIcon from 'web/lib/icons/triangle-fill-icon'
+import { Col } from '../layout/col'
+import { OUTCOME_TO_COLOR } from '../outcome-label'
+import { useSaveShares } from '../use-save-shares'
+import {
+  BinaryResolutionOrChance,
+  NumericResolutionOrExpectation,
+  FreeResponseResolutionOrChance,
+} from './contract-card'
+
+export function QuickBet(props: {
+  contract: Contract
+  setLiveUpdate: (liveUpdate: boolean) => void
+}) {
+  const { contract, setLiveUpdate } = props
+
+  const user = useUser()
+  const userBets = useUserContractBets(user?.id, contract.id)
+  const { yesFloorShares, noFloorShares, yesShares, noShares } = useSaveShares(
+    contract as FullContract<CPMM | DPM, Binary>,
+    userBets
+  )
+  // For some reason, Floor Shares are inverted for non-BINARY markets
```

Comment:
`useSaveShares` is only for binary markets, currently? yes/no floor shares is probably nonsense for non-binary

You could update the hook to save shares for FR and Numeric.

Reply:
Hm, well it's doing _something_, specifically it's useful for figuring out that there's a correct bet. Putting a TODO for now to investigate more in the future.